src/winkerberos.c

```diff
@@ -475,7 +609,15 @@ sspi_client_step(PyObject* self, PyObject* args) {
         return NULL;
     }
 
-    result = auth_sspi_client_step(state, challenge);
+    if (pychan_bindings != NULL) {
+        if (!PyCObject_Check(pychan_bindings)) {
+            PyErr_SetString(PyExc_TypeError, "Expected a SecPkgContext_Bindings object");
+            return NULL;
+        }
+        sec_pkg_context_bindings = (SecPkgContext_Bindings *)PyCObject_AsVoidPtr(pychan_bindings);
```

Comment:
The call to PyCapsule_GetPointer can fail, returning NULL:

https://docs.python.org/3.1/c-api/capsule.html#PyCapsule_GetPointer

It's generally good practice to always check return values in C.